### PR TITLE
Add support for T5 for deepspeed-inference

### PR DESCRIPTION
**Pull request by HeyangQin**
This is the PR to add support to T5. Currently it is still work in progress. A lot of the codes are adapted from https://github.com/microsoft/DeepSpeed/pull/2451

### CODE REVIEW

**Comment by alexcoca** [comment]
Hi @HeyangQin, is there any update on the timeline of supporting T5 using this work?

File: deepspeed/model_implementations/encoder_decoder/ds_t5.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        encoder_attention_mask=None,
+        encoder_decoder_position_bias=None,
+        layer_head_mask=None,
+        cross_attn_layer_head_mask=None,
+        past_key_value=None,
+        use_cache=False,
+        output_attentions=False,
+        return_dict=True,
+    ): # Adapted from https://github.com/huggingface/transformers/blob/main/src/transformers/models/t5/modeling_t5.py
+        print("Using DS_T5Container.forward")
```
Comment (baptistejamin) [inline deepspeed/model_implementations/encoder_decoder/ds_t5.py:46]: There is an extra log here

**Comment by alexcoca** [comment]
@HeyangQin @loadams I take it that this is not on your roadmap?

**Comment by HeyangQin** [comment]
Hello @alexcoca. T5 support should already be live with https://github.com/microsoft/DeepSpeed/pull/2962. Please feel free to let us know if you need more information

**Pull request closed without merging**